frontend/projects/abtesting/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts

```diff
@@ -185,6 +185,11 @@ export class ExperimentDesignComponent implements OnInit, OnChanges, OnDestroy {
 
   removeConditionOrPartition(type: string, groupIndex: number) {
     this[type].removeAt(groupIndex);
+    const deletedConditionCode = this.experimentInfo.conditions.find(condition => condition.order === groupIndex+1).id;
```

Comment:
This logic seems fragile like you are finding by relying on the `groupIndex + 1`. Can you update it in a better way than this?

Also there is a formatting issue. Please fix that as well

Reply:
GroupIndex starts with 0 and order starts from 1. Resolved minor edge case in latest commit.